crates/matrix-sdk/src/client/mod.rs

```diff
@@ -521,6 +531,40 @@ impl Client {
         Ok(res.capabilities)
     }
 
+    /// Get the server version information from the federation API.
+    ///
+    /// This method calls the `/_matrix/federation/v1/version` endpoint to get
+    /// both the server name and version.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use matrix_sdk::Client;
+    /// # use url::Url;
+    /// # async {
+    /// # let homeserver = Url::parse("http://example.com")?;
+    /// let client = Client::new(homeserver).await?;
+    ///
+    /// let server_info = client.server_version().await?;
+    /// println!("Server: {}, Version: {}", server_info.server_name, server_info.version);
+    /// # anyhow::Ok(()) };
+    /// ```
+    pub async fn server_version(&self) -> HttpResult<ServerVersionInfo> {
+        let res = self.send(get_server_version::v1::Request::new()).await?;
+        
+        // Extract server info, using defaults if fields are missing
+        let server = res.server.unwrap_or_default();
+        let server_name_str = server.name.unwrap_or_else(|| "unknown".to_string());
+        let version = server.version.unwrap_or_else(|| "unknown".to_string());
+        
+        // Try to parse the server name, fallback to a safe default if parsing fails
+        let server_name = server_name_str
+            .try_into()
+            .unwrap_or_else(|_| "unknown.server".try_into().unwrap());
```

Comment:
As per the other comment: this isn't required, as the `server_name_str` doesn't represent a homeserver name.